src/shared/scrapers/US/TX/harris-county.js

```diff
@@ -0,0 +1,146 @@
+//     `parentClassName = 'rowHeaders'` for row labels
+//     `parentClassName = 'bodyCells'` for actual body of the table
+function _getTableVals($, parentClassName) {
+  const elements = $(`div.${parentClassName} > div > div > div.pivotTableCellWrap`);
+  return elements.toArray().map(a =>
+    $(a)
+      .text()
+      .trim()
+  );
+}
+
+function _getColumnData($, n) {
+  const col = $('div.bodyCells > div > div > div').eq(n);
+  const cells = $(col)
+    .find('div.pivotTableCellWrap')
+    .toArray();
+  return cells.map(a => parseInt($(a).text(), 10));
+}
+
+// Get the values from a PowerBI pivotTable.
+const getDataFromPivotTable = $ => {
+  // check that we have Date
+  const indexNameVal = $('div.corner > div > div.pivotTableCellWrap')
+    .text()
+    .trim();
+  if (indexNameVal.trim() !== 'Date') {
+    throw Error(`Expected index name to be Date, found ${indexNameVal}`);
+  }
+
+  const colnames = _getTableVals($, 'columnHeaders').slice(0, 2);
+  const expectedColname = [
+    'Patients in General Beds (Suspected + Confirmed)',
+    'Patients in Intensive Care Beds (Suspected + Confirmed)'
+  ];
+  assert.equal(colnames.join(';'), expectedColname.join(';'), 'Column names');
+
+  const dates = _getTableVals($, 'rowHeaders')
+    .map(x => new Date(x))
+    .map(datetime.parse)
+    .map(x => x.toString());
+
+  const ret = {
+    colnames: _getTableVals($, 'columnHeaders'),
+    dates,
+    generalBeds: _getColumnData($, 0),
+    icuBeds: _getColumnData($, 1)
+  };
+
+  assert.equal(ret.dates.length, ret.generalBeds.length, 'general beds');
+  assert.equal(ret.dates.length, ret.icuBeds.length, 'icu beds');
+  return ret;
+};
+
+const scraper = {
+  county: 'fips:48201',
+  state: 'iso2:US-TX',
+  country: 'iso1:US',
+  url:
+    'https://app.powerbi.com/view?r=eyJrIjoiYjU5MzU4NjAtZWJjMC00MTllLTkwYjYtMzE4ODY1YjAyMGU2IiwidCI6ImI3MjgwODdjLTgwZTgtNGQzMS04YjZmLTdlMGUzYmUxMGUwOCIsImMiOjN9',
+  sources: [
+    {
+      name: 'SouthEast Texas Regional Advisory Council',
+      url: 'https://www.setrac.org/',
+      description: 'SouthEast Texas Regional Advisory Council'
+    }
+  ],
+  _counties: ['Harris County'],
+  timeseries: true,
+  aggregate: 'county',
+  type: 'json',
+  maintainers: [maintainers.sglyon],
+  scraper: {
+    '0': async function() {
+      const callback = async page => {
+        page.setDefaultTimeout(10000);
+        const icuButton = await page.waitForXPath("//span[text()='Hospital/COVID Census']/..");
+        icuButton.click();
+
+        const harrisSelector = await page.waitForXPath("//div[@aria-label='Harris']");
+        harrisSelector.click();
+
+        // use context menu to open table version of chart
+        await (await page.$$('div.visual-lineChart'))[0].click({ button: 'right' });
+        const tableButton = await page.waitForXPath("//drop-down-list-item[//h6[text()='Show as a table']]", {
+          visible: true
+        });
+
+        await page.waitFor(2000);
```

Comment:
Anti pattern. Can we wait for some element instead? Otherwise we’re probably waiting longer than we need to, and possibly sometimes not long enough.

Reply:
Right, good call.  Leaving for now.